### ISSUE TITLE
Restart service automatically after restart limit reached and limit interval elapsed

### ISSUE THREAD

**Issue by nebulade** [systemd/systemd#2416]
Hi, Is there any way I can make systemd restart a service (with `Restart=always`) after the `StartLimitInterval` elapsed, when the service entered the failed state due to the set limits?

From the docs:
_Note that units which are configured for Restart= and which reach the start limit are not attempted to be restarted anymore; however, they may still be restarted manually at a later point, from which point on, the restart logic is again activated_

So basically I would like to automate the _restarted manually at a later point_. This is useful in our case, if we have a service which may crash due to intermediate issues, which can go away after some period of time, but we do not want to keep restarting the service up to that point frequently.

Thanks.

**Comment by poettering** [comment]
Humm? Not sure I grok what you want to do. If you want to keep restarting forever, then set StartLimitInterval= to 0. But if you set it to anything but 0, then we eventually stop restarting the service, that's the whole point of the concept! I mean, if you want to disable the effect of StartLimitInterval= after a while, then there's really no point in useing it at all, and just setting it to 0...

If you want to increase the time we wait between automatic restarting use RestartSec=.

**Comment by fake-name** [comment]
Out of curiosity, what's the logic behind giving up on restarting a service when `Restart=always` is specified?

Naively, I'd assume a configuration directive saying "I want to always restart this" to, you know, always be restarted. Even if it fails however many times in whatever timeframe.

**Comment by koallen** [comment]
> Out of curiosity, what's the logic behind giving up on restarting a service when `Restart=always` is specified?
> 
> Naively, I'd assume a configuration directive saying "I want to always restart this" to, you know, always be restarted. Even if it fails however many times in whatever timeframe.
> 
> The assortment of other issues above that reference this one at least seem to indicate that the current behaviour is confusing.

`Restart=` only determines whether a service should be restarted or not, see https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=. It has nothing to do with the restart schedules.

**Comment by koallen** [comment]
@poettering I think that the OP was trying to ask is whether we can do something like

```
                                                      StartLimitInterval passed
                                                                  v
failed ---5s---> start & failed ---5s---> start & failed ---1h--->start again
                                                         ^
                                                hit StartLimitBurst
```

so, within each interval, we retry `StartLimitBurst` times and stop, but retry again once the `StartLimitInterval` (or `StartLimitIntervalSec`) has passed.

Based on your explanation and my own testing, I'm afraid this is not the case?

**Comment by fake-name** [comment]
> Restart= only determines whether a service should be restarted or not, see https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=. It has nothing to do with the restart schedules.

I understand that's what it currently *does*, my point is there's no possible interpretation of what `always` means that in any way matches the current implemented behavior.  

Either the `always` option needs to be named something that matches it's actual behavior, or always should actually mean *always*.
